policies/releasestrat.html

```diff
@@ -55,65 +56,20 @@ <h5>
 	  fixes. Before that, bug and security fixes will be applied
 	  as appropriate.</p>
 
-	  <p>The next version of OpenSSL will be 1.1.1 which will be an LTS release.
-	  This is currently in development and has a primary focus of implementing
-	  TLSv1.3. The RFC for TLSv1.3 has not yet been published by the IETF.
-	  OpenSSL 1.1.1 will not have its final release until that has happened.</p>
-
-	  <p>The draft release timetable for 1.1.1 is as follows. This may be
-          amended at any time as the need arises.</p>
-
+	  <p>The following stability rules apply to all changes made to code
+	  targetted for a version after 1.1.1:</p>
 	  <ul>
-	    <li>13th February 2018, alpha release 1 (pre1)</li>
-	    <li>27th February 2018, alpha release 2 (pre2)</li>
-	    <li>20th March 2018, beta release 1 (pre3)
-              <ul>
-	        <li>OpenSSL_1_1_1-stable created (feature freeze)</li>
-	        <li>master becomes basis for 1.1.2 or 1.2.0 (TBD)</li>
-	      </ul>
-	    <li>3rd April 2018, beta release 2 (pre4)</li>
-	    <li>17th April 2018, beta release 3 (pre5)</li>
-	    <li>1st May 2018, beta release 4 (pre6)</li>
-	    <li>29th May 2018, beta release 5 (pre7)</li>
-	    <li>19th June 2018, beta release 6 (pre8)</li>
-	    <li>Release readiness check following pre8 release (new release
-	        cycles added if required)</li>
+	    <li>No existing public interface can be removed or modified until its
+	    replacement has been in place in an LTS stable release and the original
+	    function has been documented as deprecated for at least 5 years.</li>
+	    <li>When structures are made opaque, any newly required accessor
+	    macros are added in a feature release of the extant LTS release
+	    and all supported intermediate successor releases (not in security-fix
+	    only mode). Non-trivial accessors may be static inline functions, that
+	    compile to new code in the calling application.</li>
+	    <li>No cosmetic function renaming for consistency.</li>
```

Comment:
If all that's happening is a name change, like the changes from `..._create()` to `..._new()` for some functions between 1.0.2 and 1.1.0, then the disruption caused to existing software dwarfs any benefit from the renaming. We failed to provide a smooth transition to the new names, and though the old names did not fit into the new naming pattern, they did the job.  Such renaming would have to be handled with care (the above 5 year deprecation timeline, backport of new names to old releases, ...) and the pain would have to be worth the effort.  Perhaps sometimes this will be worth the trouble, but often the old names are good enough.

Reply:
In light of the 5-year timeline, this would not as much be "replacing" as "adding new APIs that match the naming scheme, but providing a shim layer so the old APIs still work".

Reply:
Renaming functions for consistency hurts all existing users of the existing name, but it helps all the future users coming to OpenSSL, and also helps the OpenSSL maintainers if it can cut down the number of questions and bug reports by confused users in the long run. I think modernizing the API is well worth doing, as long as the transition is smooth, slow and ordered. The proposed 5 year deprecation cycle seems reasonable to me.